src/main/java/service/impl/ReportServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package service.impl;
+
+import database.Storage;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import model.Transaction;
+import service.ReportService;
+
+public class ReportServiceImpl implements ReportService {
+    private static final String WORDS_SEPARATOR = ",";
+    private static final String NEW_LINE = "\n";
+    private static final String HEAD_OF_FILE = "fruit,quantity\n";
+
+    @Override
+    public void createReport(List<Transaction> transactionServiceList) {
+        for (Transaction transactionService : transactionServiceList) {
+            String fruit = transactionService.getFruitName();
+            int value = transactionService.getFruitValue();
+            transactionService.getFruitOperationType()
+                    .doTransaction(fruit, value);
+        }

```

Comment:
It’s not appropriate logic for ReportService. You should move it to TransactionProcessor. Also, pay attention to variable names. 
transactionServiceList --> transactions
transactionService --> transaction